packages/web-app/src/modules/earn-views/components/AllMachines/AllMachines.tsx

```diff
@@ -107,6 +110,52 @@ const _AllMachines = ({ classes }: Props) => {
     window.location.href = 'https://support.salad.com/article/414-how-to-find-your-salad-machine-id'
   }
 
+  const handleDropdownChange = ({ value }: DropdownOption) => {
+    switch (value) {
+      case MachinesTableDropdownOptionValue.SelectAll: {
+        setSelectedMachineIds(generatedMockedMachines.reduce((acc, machine) => ({ ...acc, [machine.id]: true }), {}))
+        break
+      }
+      case MachinesTableDropdownOptionValue.SelectAllInPage: {
+        const beginningIndex = (currentPageNumber - 1) * itemsPerPageAmount
+        const endingIndex = currentPageNumber * itemsPerPageAmount
+        const machinesToSelect = generatedMockedMachines
+          .slice(beginningIndex, endingIndex)
+          .reduce((acc, machine) => ({ ...acc, [machine.id]: true }), {})
+
+        setSelectedMachineIds((previousSelectedMachineIds) => ({
+          ...previousSelectedMachineIds,
+          ...machinesToSelect,
+        }))
+
+        break
+      }
+
+      case MachinesTableDropdownOptionValue.DeselectAll: {
+        setSelectedMachineIds({})
+        break
+      }
+
+      case MachinesTableDropdownOptionValue.DeselectAllInPage: {
+        const beginningIndex = (currentPageNumber - 1) * itemsPerPageAmount
+        const endingIndex = currentPageNumber * itemsPerPageAmount
+        const machinesToDeselect = generatedMockedMachines
+          .slice(beginningIndex, endingIndex)
+          .reduce((acc, machine) => ({ ...acc, [machine.id]: false }), {})
```

Comment:
let's keep it completely aligned with ` usePagination()` hook values. 

```suggestion
        const machinesToDeselect = generatedMockedMachines
          .slice(lowestItemNumberOnPage - 1, highestItemNumberOnPage)
          .reduce((acc, machine) => ({ ...acc, [machine.id]: false }), {})
```